libminifi/src/core/state/nodes/RepositoryMetricsSourceStore.cpp

```diff
@@ -0,0 +1,103 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "core/state/nodes/RepositoryMetricsSourceStore.h"
+
+namespace org::apache::nifi::minifi::state::response {
+
+RepositoryMetricsSourceStore::RepositoryMetricsSourceStore(std::string name) : name_(std::move(name)) {}
+
+void RepositoryMetricsSourceStore::setRepositories(const std::vector<std::shared_ptr<core::RepositoryMetricsSource>> &repositories) {
+  repositories_ = repositories;
+}
+
+void RepositoryMetricsSourceStore::addRepository(const std::shared_ptr<core::RepositoryMetricsSource> &repo) {
+  if (nullptr != repo) {
+    repositories_.push_back(repo);
+  }
+}
+
+std::vector<SerializedResponseNode> RepositoryMetricsSourceStore::serialize() const {
+  std::vector<SerializedResponseNode> serialized;
+  for (const auto& repo : repositories_) {
+    SerializedResponseNode parent;
+    parent.name = repo->getRepositoryName();
+    SerializedResponseNode is_running;
+    is_running.name = "running";
+    is_running.value = repo->isRunning();
+
+    SerializedResponseNode is_full;
+    is_full.name = "full";
+    is_full.value = repo->isFull();
+
+    SerializedResponseNode repo_size;
+    repo_size.name = "size";
+    repo_size.value = repo->getRepositorySize();
+
+    SerializedResponseNode max_repo_size;
+    max_repo_size.name = "maxSize";
+    max_repo_size.value = repo->getMaxRepositorySize();
+
```

Comment:
You can use designated initializers to describe the same data structure in a simpler way. Example from AgentInformation.h:
```cpp
    std::vector<SerializedResponseNode> serialized = {
        {.name = "identifier", .value = AgentBuild::BUILD_IDENTIFIER},
        {.name = "agentType", .value = "cpp"},
        {.name = "buildInfo", .children = {
            {.name = "flags", .value = AgentBuild::COMPILER_FLAGS},
            {.name = "compiler", .value = AgentBuild::COMPILER},
            {.name = "version", .value = AgentBuild::VERSION},
            {.name = "revision", .value = AgentBuild::BUILD_REV},
            {.name = "timestamp", .value = static_cast<uint64_t>(std::stoull(AgentBuild::BUILD_DATE))}
        }}
    };
```

Reply:
Updated in f5b67d851a249ac90ffeb0ad59f2258597701a82